main.js

```diff
@@ -1,144 +1,225 @@
+    //     navBar.classList.replace("nav-bar" ,"nav-bar-dark" );
+    //     listOfCountries.classList.replace( "default" ,"elements-dark");
+    //     filter.classList.replace( "default" ,"elements-dark");
+    //     for(var i =0 ; i<cards.length ;i++){
+    //         cards[i].classList.replace( "default" ,"elements-dark");
+    //     }
+    //     localStorage.setItem("texts" , "dark-mode-texts")
+    //     localStorage.setItem("Nav" , "nav-bar-dark")
+    //     localStorage.setItem("main" , "section-dark")
+    //     localStorage.setItem("elements" , "elements-dark")
+    // }
+}
 
-        for(var i =0 ; i< cards.length ; i++){
-            cards[i].classList.replace("elements-dark" , "default");
-        }
-    }
+darkMode.addEventListener("click", getDarkMode);
 
-}
-// }
+function getCountries() {
+    let container = "";
 
-function getDarkMode(){
-    if(darkModeToggle.classList.contains('fa-solid') && navBar.classList.contains('nav-bar-dark')){
-        darkModeToggle.classList.replace("fa-solid" ,"fa-regular" );
-        darkModeToggle.classList.remove("dark-mode-texts");
-        navBar.classList.replace("nav-bar-dark" ,"nav-bar");
-        listOfCountries.classList.replace("elements-dark" , "default")
-        filter.classList.replace("elements-dark" , "default")
-        for(var i =0 ; i<cards.length; i++){
-            cards[i].classList.replace("elements-dark" , "default")
-        }
-        for(var i = 0 ; i<textInDark.length ; i++ ){
-            textInDark[i].classList.replace("dark-mode-texts" ,"text-dark")
-        }
-        main.classList.replace( "section-dark","section")
-        searchBar.classList.replace( "section-dark","section")
-        localStorage.setItem("texts" , "text-dark")
-        localStorage.setItem("Nav" , "nav-bar")
-        localStorage.setItem("main" , "section")
-        localStorage.setItem("elements" , "default")
-    }
-    else{
-        for(var i = 0 ; i<textInDark.length ; i++ ){
-            textInDark[i].classList.replace("text-dark","dark-mode-texts")
-        }
-        main.classList.replace("section" ,"section-dark");
-        searchBar.classList.replace("section" ,"section-dark");
-        darkModeToggle.classList.add("dark-mode-texts");
-        darkModeToggle.classList.replace("fa-regular" ,"fa-solid");
-        navBar.classList.replace("nav-bar" ,"nav-bar-dark" );
-        listOfCountries.classList.replace( "default" ,"elements-dark");
-        filter.classList.replace( "default" ,"elements-dark");
-        for(var i =0 ; i<cards.length ;i++){
-            cards[i].classList.replace( "default" ,"elements-dark");
-        }
-        localStorage.setItem("texts" , "dark-mode-texts")
-        localStorage.setItem("Nav" , "nav-bar-dark")
-        localStorage.setItem("main" , "section-dark")
-        localStorage.setItem("elements" , "elements-dark")
-    }
+    countries.forEach((country) => {
+        // for(let i =0 ; i< countries.length ; i++){

```

Comment:
147 : 158 => -1

foreach method is more efficient and we can also use createElement and appendChild functions for efficient code.